packages/dapp/src/providers/DexsAndBridgesInfosProvider.tsx

```diff
@@ -0,0 +1,56 @@
+import { ToolsResponse } from '@lifi/sdk';
+import axios, { AxiosResponse } from 'axios';
+import { createContext, PropsWithChildren, useContext } from 'react';
+import { useQuery } from 'react-query';
+
+export interface DexsAndBridgesInfosContextProps {
+  data: ToolsResponse;
+  isSuccess: boolean;
+}
+
+const initialData = {
+  data: {
+    exchanges: [],
+    bridges: [],
+  },
+  isSuccess: false,
+};
+
+const DexsAndBridgesInfosContext =
+  createContext<DexsAndBridgesInfosContextProps>(initialData);
+
+export const useDexsAndBridgesInfos = () => {
+  return useContext(DexsAndBridgesInfosContext);
+};
+
+export const DexsAndBridgesInfosProvider: React.FC<PropsWithChildren<{}>> = ({
+  children,
+}) => {
+  const { data, isSuccess } = useQuery(
+    ['toolsStats'],
+    async () => {
+      const apiUrl = import.meta.env.VITE_LIFI_API_URL;
+      const result = await axios<ToolsResponse>({
+        method: 'GET',
+        url: `${apiUrl}/tools`,
+      });
+      return result;
```

Comment:
We should stop using axios and remove this dependency. We have migrated to a native fetch everywhere.

Also, we don't need a context as a cache layer here and should add a custom hook with react-query in it. react-query perfectly handles caching purpose.

Reply:
For my notes: We also have axios to fetch the chains in a seperate hook. Gotta create another task to remove axios dependency on "ChainInfosProvider.tsx".

Reply:
This is the ticket: https://lifi.atlassian.net/browse/LF-3299